mne/io/nsx/tests/test_nsx.py

```
@@ -0,0 +1,143 @@
+
+    # Check annotations
+    assert len(raw.annotations) == 0
+
+    raw.load_data()
+    raw_data, times = raw[:]
+    n_channels, n_times = raw_data.shape
+    assert times.shape[0] == n_times
+    orig_data = raw_data / (raw.info['chs'][0]['cal']
+                            * raw.info['chs'][0]['range'])
+    assert np.allclose(sum(orig_data[:, 50] - 10 - np.arange(n_channels)), 76.)
+
+    assert np.allclose(orig_data[n_channels // 2, :100] - 100, np.arange(100))
+
+    data, times = raw.get_data(start=10, stop=20, return_times=True)
+    assert n_channels, 10 == data.shape
+
+    data, times = raw.get_data(start=0, stop=300, return_times=True)
+
+
+@requires_testing_data
+def test_stim_eog_misc_chs_in_nsx():
+    raw = read_raw_nsx(nsx_22_fname, stim_channel='elec127', eog=['elec126'])
+    assert raw.info['chs'][127]['kind'] == FIFF.FIFFV_STIM_CH
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=['elec127'], eog=['elec126'])
+    assert raw.info['chs'][127]['kind'] == FIFF.FIFFV_STIM_CH
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=127, eog=['elec126'])
+    assert raw.info['chs'][127]['kind'] == FIFF.FIFFV_STIM_CH
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    raw = read_raw_nsx(nsx_22_fname, stim_channel=[127], eog=['elec126'])
+    assert raw.info['chs'][127]['kind'] == FIFF.FIFFV_STIM_CH
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    stims = [ch_info['kind'] == FIFF.FIFFV_STIM_CH for ch_info
+             in raw.info['chs']]
+    assert np.any(stims)
+    assert raw.info['chs'][126]['kind'] == FIFF.FIFFV_EOG_CH
+    try:
+        raw = read_raw_nsx(nsx_22_fname, stim_channel=['elec128', 129],
+                           eog=['elec126'])
+    except ValueError as err:
+        err_msg = err.args[0]
+    assert err_msg == 'Invalid stim_channel'
```

you want to use `pytest.raises` ... in general avoid `try`/`except` pattern when you can. It will silence errors

---

Thanks for pointing that out.